### PR TITLE
Codesniffer: remove PHP 7.0 and 7.1 support

### PR DESCRIPTION
**Pull request by tbradsha**
As the title says, it removes PHP 7.0 and 7.1 support, and updates the tests accordingly.

We should add PHP 8+ tests, but that will probably be after updating to PHPUnit 10+.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Verify CI is happy, or run tests manually: `composer test-php`

### CODE REVIEW

File: projects/packages/codesniffer/Jetpack/ruleset.xml [hunk 1]
```diff
@@ -25,9 +25,6 @@
 	<rule ref="MediaWiki.WhiteSpace.EmptyLinesBetweenUse" />
 	<rule ref="MediaWiki.WhiteSpace.MultipleEmptyLines" />
 	<rule ref="MediaWiki.Usage.IsNull" />
-	<!-- Maybe in the future
```
Comment (zinigor) [inline projects/packages/codesniffer/Jetpack/ruleset.xml:28]: Maybe in another universe :)

**Review by zinigor — DISMISSED** [review]
Sweet exorcism! Also alphabetical ordering FTW 👍🏽

File: projects/packages/codesniffer/Jetpack/ruleset.xml (hunk 1 above)
Comment (tbradsha) [reply projects/packages/codesniffer/Jetpack/ruleset.xml:28]: In this case, it's redundant to another rule. :^)

**Comment by tbradsha** [comment]
> Also alphabetical ordering FTW

These are auto-generated files. That said, I wonder why the order is different...I suspect the default on my machine vs. Brad's is different. 🤔

Edit: I tried running it after using `export LC_COLLATE=C` but there were no changes when I did so. @zinigor @anomiex can you confirm the generated files on your end result in no changes in the files that are committed in this branch? Ideally generation would be consistent.

File: projects/packages/codesniffer/build-compat-rulesets.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ DIR="$(mktemp -d "${TMPDIR%/}/codesniffer-build-compat-rulesets.XXXXXXXX")"
 trap 'rm -rf "$DIR"' EXIT
 cd "$DIR"
 
-PHP_VERSIONS=( 7.0 7.1 7.2 7.3 7.4 8.0 8.1 8.2 8.3 )
+PHP_VERSIONS=( 7.2 7.3 7.4 8.0 8.1 8.2 8.3 )
```
Comment (anomiex) [inline projects/packages/codesniffer/build-compat-rulesets.sh:14]: Should we go ahead and add 8.4 right away, or save that for a followup?

**Review by anomiex — COMMENTED** [review]
> can you confirm the generated files on your end result in no changes in the files that are committed in this branch?

Adding `LC_ALL=C.UTF-8` at the top of the script makes it consistent with what you have here for me. So let's do that to make it consistent.

**Comment by anomiex** [comment]
> We should add PHP 8+ tests, but that will probably be after updating to PHPUnit 10+.

It's not PHPUnit that needs updating, it's [phpcompatibility/php-compatibility](https://packagist.org/packages/phpcompatibility/php-compatibility). I'm not sure if PHP 8.0 checks still aren't finished or if the v10 release is now waiting on 8.4 checks, but either way there's no release to update to. 🙁 https://xkcd.com/2347/

File: projects/packages/codesniffer/build-compat-rulesets.sh (hunk 2 above)
Comment (tbradsha) [reply projects/packages/codesniffer/build-compat-rulesets.sh:14]: I was figuring a separate PR, but no strong feelings.

**Review by anomiex — APPROVED** [review]
Let's do it.

**Pull request merged**